src/java.base/share/classes/java/util/zip/InflaterOutputStream.java

```diff
@@ -95,6 +117,11 @@ public InflaterOutputStream(OutputStream out, Inflater infl) {
      * Creates a new output stream with the specified decompressor and
      * buffer size.
      *
+     * @apiNote
+     * {@linkplain #close() Closing} this output stream
+     * {@linkplain ##decompressor-usage will not close} the given
+     * {@linkplain Inflater decompressor}.
```

Comment:
This can be normative too.

Reply:
Thank you Alan for those inputs. I've updated the PR to make this normative in all the 4 class' constructors.